Style Guid and Diataxis Review: Streams II (#215)
iotaledger/streams: documentation/docs/libraries/rust/getting_started.md

---
description: Getting started with Rust in IOTA Streams.
image: /img/logo/iota_mark_light.png
keywords:
- api
- Rust
- getting started
- reference
---

# Getting Started

> Streams requires an asynchronous runtime environment to be set, so we suggest using [tokio](https://docs.rs/tokio/latest/tokio/). Streams also uses [anyhow](https://docs.rs/anyhow/latest/anyhow/) for error handling, so projects can use `anyhow::Result` and `anyhow::Error` for easier integration. 

Reviewer: ```suggestion
Streams requires that you set an asynchronous runtime environment.  We suggest using [tokio](https://docs.rs/tokio/latest/tokio/). Streams also uses [anyhow](https://docs.rs/anyhow/latest/anyhow/) for error handling, so projects can use `anyhow::Result` and `anyhow::Error` for easier integration. 
```

```

## Next Steps

Now you can gather users to subscribe to the channel and generate branches to specify access control for publishers and subscribers via keyload messages.  

### Subscription

> To subscribe to a channel, subscribers create a subscribe message that is linked to the channel announcement message. The message link should then be provided to the author. This allows the author to handle the subscription message and use the public key of the subscriber for access control and validation purposes.

Reviewer: ```suggestion
To subscribe to a channel, subscribers create a subscribe message linked to the channel announcement message. The message link should then be provided to the author. This allows the author to handle the subscription message and use the subscriber's public key for access control and validation purposes.
```
Who or what should provide the message to the author?

```

#### Author

```
// Process subscriber link 
let sub_link = Address::from_str("SUBSCRIPTION_LINK")?;
author.receive_subscribe(&sub_link).await?;
```

### Keyloads 

> Keyload messages are used as an access control mechanism for a branch. A random key is generated and masked within the message using the public keys or pre-shared keys included in them. This allows the author to specify which subscribers have access to which branches. There are two ways to send a keyload:

Reviewer: ```suggestion
Keyload messages are an access control mechanism for a branch. A random key is generated and masked within the message using its public keys or pre-shared keys. This allows the author to specify which subscribers have access to which branches. There are two ways to send a keyload:
```
